src/library/basetracktablemodel.cpp

```diff
@@ -618,12 +619,23 @@ QVariant BaseTrackTableModel::roleValue(
             return QString("(%1)").arg(timesPlayed);
         }
         case ColumnCache::COLUMN_LIBRARYTABLE_DATETIMEADDED:
-        case ColumnCache::COLUMN_LIBRARYTABLE_LAST_PLAYED_AT:
         case ColumnCache::COLUMN_PLAYLISTTRACKSTABLE_DATETIMEADDED:
             VERIFY_OR_DEBUG_ASSERT(rawValue.canConvert<QDateTime>()) {
                 return QVariant();
             }
             return mixxx::localDateTimeFromUtc(rawValue.toDateTime());
+        case ColumnCache::COLUMN_LIBRARYTABLE_LAST_PLAYED_AT: {
+            QDateTime lastPlayedAt;
+            if (rawValue.type() == QVariant::String) {
+                // column value
+                lastPlayedAt = sqlite::readGeneratedTimestamp(rawValue);
```

Comment:
The current state is that "2020-12-23T23:41:50.294" format is also written as string into the Library. 
That fails here. So we need to delay this PR and its DB repair code until this issue is solved. 
Else users will not see a time stamp for recently played tracks.

Reply:
??? This PR fixes all formatting issues and fixes the wrong values in the DB. I don't understand what you ask here for.

Reply:
Oh sorry, I was confused, because I have used a unpatched version after the new library migration. 
I have now tested again and I can't see any T splitter formats anymore. The offset issue is also entirely solved.